engine/graphics/src/vulkan/graphics_vulkan.cpp

```diff
@@ -2474,6 +2474,27 @@ namespace dmGraphics
         vkCreatePipelineLayout(context->m_LogicalDevice.m_Device, &vk_layout_create_info, 0, &program->m_Handle.m_PipelineLayout);
     }
 
+    static void ResolveSamplerTextureUnits(Program* program, const dmArray<ShaderResourceBinding>&  texture_resources)
```

Comment:
This is needed for the "constructed" samplers from the pipeline. Since we now have to split combined samplers into texture + sampler, we need to be able to match the sampler with the texture, so that we can assign the same texture unit to both.